### DISCUSSION TITLE
Appropriate CMakeLists.txt for integration in VCPkg

### DISCUSSION THREAD

**Discussion by drathin4** [microsoft/vcpkg#27500] [Q&A]
This question is about `CMakeLists.txt` file in the context of `VCPkg`. I'm a novice in `CMake` scripting and hence asking this question.

I have a custom library `fleck` that I want to integrate in `VCPkg`. The existing `CMakeLists.txt` of `fleck` is working, but it is meant for standalone build process. Now I want to tailor it for integration in `VCPkg`. I already had prior communication in the link below, although it was based on `vcpkg-registry`: 
[https://github.com/microsoft/vcpkg/discussions/27308](url)

Currently, I'm having a problem with the build process. `Eigen3, PCL1.3, Qt5, VectorClass` are the dependencies of this library. CMake cannot locate the header files of the dependencies (`"vector.h" from vectorclass` and `Qt* headers from Qt5::Widgets`). I tried using either of the commands in `CMakeLists.txt`, but neither of them are working.
`
#target_include_directories( ${SOURCE_PATH} ${Qt5Widgets_INCLUDES} ${EIGEN3_INCLUDE_DIR} ${PCL_INCLUDE_DIRS} )
#target_link_libraries(${PROJECT_NAME} PRIVATE vectorclass::vectorclass Eigen3::Eigen Qt5::Widgets ${PCL_LIBRARIES})
`

I'm using the `install` command in `CMakeLists.txt` as it is used for other libraries. I'm attaching the 2 files: `CMakeLists.txt & install-x64-windows-dbg-out.log` for your perusal:

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9879916/install-x64-windows-dbg-out.log)
[CMakeLists.txt](https://github.com/microsoft/vcpkg/files/9879917/CMakeLists.txt)

**Comment by JavierMatosD** [comment]
Hi @drathin4, is this discussion related to this one: https://github.com/microsoft/vcpkg/discussions/27573?

**Reply by drathin4** [reply]
I'll hold this discussion for the time being, till #27573 is resolved. If needed, I'll ask for help.

**Reply by JavierMatosD** [reply]
Hi @drathin4, I believe #27573 has been resolved. Do you still require assistance here?

**Reply by drathin4** [reply]
This discussion can be deleted from your end as the problem is solved and the discussion doesn't even contain any useful information in it.
